dali/python/nvidia/dali/external_source.py

```diff
@@ -424,8 +424,6 @@ def __init__(
         self._batch_info = batch_info
 
         self._spec.AddArg("device", device)
-        if dtype is not None:
```

Comment:
This caused the problem. C++ implementation supports only single input and when a list of types was provided, it complained about incorrect argument type.